scripts/dashboard/validate_dashboards_format.py

```diff
@@ -0,0 +1,153 @@
+
+import json
+import os
+import sys
+import yaml
+
+class MetadataFormattingError(Exception):
+  '''Raised when data in metadata file does not have expected format'''
+  pass
+
+class JsonFormattingError(Exception):
+  '''Raised when data in json file or file name does not have expected format'''
+  pass
+
+class MetadataInJsonMisMatchError(Exception):
+  '''Raised when data in metadata file does not have matching data in a json file'''
+  pass
+
+class JsonInMetadataMisMatchError(Exception):
+  '''Raised when data in json file does not have matching data in a metadata file'''
+  pass
+
+def mapPaths(paths):
+  '''Takes in list of paths and returns dictionary of metadata and dashboard json data, 
+  organized by directories'''
+  map = {}
+
+  for path in paths:
+    pathParts = path.split('/')
+
+    # filter out non-dashboards files
+    if not 'dashboards' == pathParts[1]:
+        continue
+
+    if not pathParts[2] in map.keys():
+      map[pathParts[2]] = {
+        'metadataIds': {},
+        'metadataNames': {},
+        'jsonIds': {},
+        'jsonNames': {},
+        'jsonFiles': []
+      }
+
+    if pathParts[3].split('.')[-1] == "json":
+      map[pathParts[2]]['jsonFiles'].append(pathParts[3])
+
+  return map
+
+def get_sample_dashboards_json(path):
+  '''Checks if file is in json format and returns data in dictionary form'''
+  with open(path) as f:
+    try:
+      dashboards_dict = json.load(f)
+      dashboards_json = json.dumps(dashboards_dict)
+
+      return dashboards_dict
+
+    except JsonFormattingError:
+      print("{} content could not be loaded".format(path))
+
+def check_json_file_name(path, file_name_parts):
+  '''Checks if json file name is in the proper format'''
+  if len(file_name_parts) != 2:
+    raise JsonFormattingError("{} file name not in <name>.json format".format(path))
+
+def mapJsonFiles(directory, pathMap):
+  '''Stores json data into dictionary'''
+  for jsonFile in pathMap[directory]['jsonFiles']:
+    jsonPath = os.path.join('.', 'dashboards', directory, jsonFile)
+    dashDict = get_sample_dashboards_json(jsonPath)
+
+    if not 'displayName' in dashDict.keys():
+      raise JsonFormattingError("{} is missing displayName field".format(jsonFile))
+
+    pathMap[directory]['jsonNames'][dashDict['displayName']]=''
+
+    fileNameParts = jsonFile.split('.')
+    check_json_file_name(jsonFile, fileNameParts)
+
+    pathMap[directory]['jsonIds'][fileNameParts[0]]=''
+
+def check_metadata_entries(path, sample_dashboard):
+  '''Check if metadata file fields are as expected'''
+  required_fields = {"category", "id", "display_name", "description"}
+  missing_fields = required_fields - sample_dashboard.keys()
+  if missing_fields:
+    raise MetadataFormattingError("{} missing {}".format(path, missing_fields))
+
+def check_metadata(directory, pathMap):
+  '''Checks metadata file in directory for correct fields and matching data in 
+  json files within the same directory'''
```

Comment:
This sentence is difficult to parse. I think this might read better as a list, e.g.

```
Make following assertions about metadata.yam files:
* At least 1 sample dashboard exists
* Each dashboard entry has all required fields
* Each dashboard entry has a corresponding json template with same name & id
```

Reply:
Would also be good to add a line that it mutates pathMap

Reply:
done.